lib/formats/gzip.js

```diff
@@ -97,52 +99,41 @@ function addTarEntry(pack, options, data = null) {
 }
 
 async function compressAttachments(apos, pack, attachments = {}) {
+
+  const copyOut = Promise.promisify(apos.attachment.uploadfs.copyOut);
+
   await addTarEntry(pack, {
     name: 'attachments/',
     type: 'directory'
   });
 
-  const chunkedPromises = chunkPromises(getPromises(), 5);
   let attachmentError = false;
 
-  for (const chunk of chunkedPromises) {
-    const results = await Promise.allSettled(chunk);
-    if (results.some(({ status }) => status === 'rejected')) {
-      attachmentError = true;
-    }
-  }
+  await Promise.map(Object.entries(attachments), processOneAttachment, {
+    concurrency: 5
+  });
 
   return { attachmentError };
 
-  function getPromises() {
-    return Object.entries(attachments).map(([ name, url ]) => {
-      return new Promise((resolve, reject) => {
-        apos.http.get(url, { originalResponse: true })
-          .then((res) => {
-            res.body.on('error', reject);
-
-            res
-              .buffer()
-              .then((buffer) => {
-                addTarEntry(pack, { name: `attachments/${name}` }, buffer)
-                  .then(resolve)
-                  .catch(reject);
-              })
-              .catch(reject);
-          })
-          .catch(reject);
+  async function processOneAttachment([ name, url ]) {
+    const temp = apos.attachment.uploadfs.getTempPath() + '/' + apos.util.generateId();
+    try {
+      await copyOut(url, temp);
```

Comment:
We're doing this because it works even when *.localhost isn't pointing to localhost, but also because it avoids loading a big buffer into memory